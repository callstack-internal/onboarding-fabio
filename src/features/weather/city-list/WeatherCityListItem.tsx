import React from 'react';
import {Image, Pressable, StyleSheet, Text, View} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
import {CityModel} from '../../../data/model/city/CityModel';
import Colors from '../../../styles/Colors';

type Props = {
  item: CityModel;
  onPress?: (item: CityModel) => void;
};

const WeatherCityListItem = ({item, onPress}: Props) => {
  const _onPress = () => {
    onPress?.(item);
  };

  return (
    <Pressable accessibilityRole={'button'} onPress={_onPress}>
      <SafeAreaView style={styles.rootContainer} edges={['left', 'right']}>
        <Image
          style={styles.image}
          source={{uri: item.iconURL}}
          accessibilityIgnoresInvertColors
        />

        <View style={styles.detailsContainer}>
          <Text style={styles.detailsNameLabel} numberOfLines={1}>
            {item.name}
          </Text>
          <Text style={styles.detailsWeatherLabel} numberOfLines={1}>
            {item.weather}
          </Text>
        </View>

        <View style={styles.chipContainer}>
          <Text style={styles.chipLabel} numberOfLines={1}>
            {`${item.temperature} Â°F`}
          </Text>
        </View>

        {onPress && (
          <MaterialIcon
            style={styles.chevronIcon}
            name={'chevron-right'}
            size={24}
            color={Colors.black}
          />
        )}
      </SafeAreaView>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  rootContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: Colors.white,
    padding: 20,
  },

  image: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: Colors.lightGray,
  },

  detailsContainer: {
    flex: 1,
    marginHorizontal: 20,
  },
  detailsNameLabel: {
    fontSize: 14,
    color: Colors.black,
  },
  detailsWeatherLabel: {
    fontSize: 12,
    fontWeight: 'bold',
    color: Colors.gray,
    marginTop: 4,
  },

  chipContainer: {
    height: 30,
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 15,
    backgroundColor: Colors.lightGray,
  },
  chipLabel: {
    fontSize: 14,
    color: Colors.black,
  },

  chevronIcon: {
    marginLeft: 10,
  },
});

export default WeatherCityListItem;
