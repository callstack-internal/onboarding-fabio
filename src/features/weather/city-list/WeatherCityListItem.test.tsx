import React from 'react';
import renderer from 'react-test-renderer';
import WeatherCityListItem from './WeatherCityListItem';

test('should render correctly', () => {
  const tree = renderer
    .create(
      <WeatherCityListItem
        item={{
          id: 2988507,
          name: 'Paris',
          weather: 'Mist',
          temperature: 32.47,
          pressure: 1031,
          humidity: 95,
          windSpeed: 5.75,
          cloudCover: 100,
          iconURL: 'https://openweathermap.org/img/wn/50d.png',
        }}
        onPress={() => {}}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('should render a different state correctly', () => {
  expect(
    <WeatherCityListItem
      item={{
        id: 2988507,
        name: 'Paris',
        weather: 'Mist',
        temperature: 32.47,
        pressure: 1031,
        humidity: 95,
        windSpeed: 5.75,
        cloudCover: 100,
        iconURL: 'https://openweathermap.org/img/wn/50d.png',
      }}
    />,
  ).toMatchDiffSnapshot(
    <WeatherCityListItem
      item={{
        id: 3117735,
        name: 'Madrid',
        weather: 'Clear',
        temperature: 47.44,
        pressure: 1025,
        humidity: 46,
        windSpeed: 6.91,
        cloudCover: 0,
        iconURL: 'https://openweathermap.org/img/wn/01d.png',
      }}
      onPress={() => {}}
    />,
  );
});
