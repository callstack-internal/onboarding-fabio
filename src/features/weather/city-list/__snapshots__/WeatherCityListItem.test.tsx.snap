// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a different state correctly 1`] = `
"Snapshot Diff:
- <WeatherCityListItem item={{"cloudCover": 100, "humidity": 95, "iconURL": "https://openweathermap.org/img/wn/50d.png", "id": 2988507, "name": "Paris", "pressure": 1031, "temperature": 32.47, "weather": "Mist", "windSpeed": 5.75}} />
+ <WeatherCityListItem item={{"cloudCover": 0, "humidity": 46, "iconURL": "https://openweathermap.org/img/wn/01d.png", "id": 3117735, "name": "Madrid", "pressure": 1025, "temperature": 47.44, "weather": "Clear", "windSpeed": 6.91}} onPress={[Function onPress]} />

@@ -48,11 +48,11 @@
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        source={
          Object {
-           "uri": "https://openweathermap.org/img/wn/50d.png",
+           "uri": "https://openweathermap.org/img/wn/01d.png",
          }
        }
        style={
          Object {
            "backgroundColor": "#D3D3D3",
@@ -77,11 +77,11 @@
              "color": "#000000",
              "fontSize": 14,
            }
          }
        >
-         Paris
+         Madrid
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
@@ -90,11 +90,11 @@
              "fontWeight": "bold",
              "marginTop": 4,
            }
          }
        >
-         Mist
+         Clear
        </Text>
      </View>
      <View
        style={
          Object {
@@ -113,10 +113,33 @@
              "color": "#000000",
              "fontSize": 14,
            }
          }
        >
-         32.47 °F
+         47.44 °F
        </Text>
      </View>
+     <Text
+       allowFontScaling={false}
+       selectable={false}
+       style={
+         Array [
+           Object {
+             "color": "#000000",
+             "fontSize": 24,
+           },
+           Object {
+             "marginLeft": 10,
+           },
+           Object {
+             "fontFamily": "Material Icons",
+             "fontStyle": "normal",
+             "fontWeight": "normal",
+           },
+           Object {},
+         ]
+       }
+     >
+       
+     </Text>
    </RNCSafeAreaView>
  </View>"
`;

exports[`should render correctly 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <RNCSafeAreaView
    edges={
      [
        "left",
        "right",
      ]
    }
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "padding": 20,
      }
    }
  >
    <Image
      accessibilityIgnoresInvertColors={true}
      source={
        {
          "uri": "https://openweathermap.org/img/wn/50d.png",
        }
      }
      style={
        {
          "backgroundColor": "#D3D3D3",
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      >
        Paris
      </Text>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#808080",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginTop": 4,
          }
        }
      >
        Mist
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#D3D3D3",
          "borderRadius": 15,
          "height": 30,
          "paddingHorizontal": 12,
          "paddingVertical": 6,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "color": "#000000",
            "fontSize": 14,
          }
        }
      >
        32.47 °F
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#000000",
            "fontSize": 24,
          },
          {
            "marginLeft": 10,
          },
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </RNCSafeAreaView>
</View>
`;
